torch/csrc/jit/script/resolver.h

```diff
@@ -29,6 +29,9 @@ struct Resolver {
       const std::string& name,
       Function& m,
       const SourceRange& loc) const = 0;
+
+  // Resolve `name` to a TypePtr.
+  virtual TypePtr resolveType(const std::string& name) const = 0;
```

Comment:
Once there are qualified names we need to be careful of `name` vs `qualname` vs `basename` to keep things clear

Reply:
My vote is to use the QualName class when that change is made. Separate strings are easy to lose track of.